app/vmctl/influx/parser.go

```diff
@@ -36,6 +36,72 @@ func parseResult(r influx.Result) ([]queryValues, error) {
 	return qValues, nil
 }
 
+// parseResultCheckTags check rows return by InfluxDB and remove rows not belong to current series.
+// See: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7301.
+func parseResultCheckTags(s *Series, r influx.Result) ([]queryValues, error) {
+	if len(r.Err) > 0 {
+		return nil, fmt.Errorf("result error: %s", r.Err)
+	}
+
+	// This map contains all the desired column of the query result for current series.
+	// If a column (`Columns[i]` in influx.Result) not exists in map, this column must be declared by other series.
+	// A row (`values[i]` in influx.Result) is considered invalid if its unwanted column value is not null.
+	wantedColumns := make(map[string]bool)
```

Comment:
I think we can make a map with the length of the `s.LabelPairs`.

Reply:
Thanks for the catch